code/modules/materials/Mat_MaterialProcs.dm

```diff
@@ -738,3 +738,54 @@ triggerOnEntered(var/atom/owner, var/atom/entering)
 		var/list/color = rgb2num(owner.material.getColor())
 		light_c = owner.AddComponent(/datum/component/loctargeting/sm_light, color[1], color[2], color[3], 255 * 0.33)
 		light_c.update(1)
+
+/datum/materialProc/radioactive_temp
+	max_generations = -1
+
+	execute(var/atom/owner, var/temp)
+		if (isobj(owner) && temp > 500 KELVIN)
+			/// How much this individual thing will affect the decay of plasma on its tile. Considers item amounts only for now
+			var/bulkness_multiplier = 1
+			var/obj/O = owner
+			if (isitem(O))
+				var/obj/item/I = O
+				bulkness_multiplier = I.amount
+				if (I.amount <= 0)
+					return // I just have a feeling this will manage to happen and fuck things up
+			if (!O.material.hasProperty("radioactive"))
+				return
+			var/radioactivity = O.material.getProperty("radioactive")
+			var/n_radioactivity = O.material.getProperty("n_radioactive")
+			if (O.material.hasProperty("n_radioactive"))
+				n_radioactivity = O.material.getProperty("n_radioactive")
+			var/turf/T = locate(O.x, O.y, O.z)
```

Comment:
```suggestion
			var/turf/T = get_turf(O)
```
Unless you're meaning this to fail inside containers, in which case `O.loc` would better I think.

Reply:
You probably shouldn't be doing it like this at all. See the molitz temp react for an example. Basically you want `owner.return_air()` instead of this.